### ISSUE TITLE
RFC: Removing potentially unused feature: 'Preserve contents of playing segment when unsynced'

### ISSUE THREAD

**Issue by Julusian** [Sofie-Automation/sofie-core#1103]
## About Me
This RFC is posted on behalf of the NRK.

## Background

At NRK, we are planning to do some refactoring of the Ingest logic to match the Playout change https://github.com/nrkno/sofie-core/pull/1035.
We have found one feature whose current implementation will not easily translate across to this new structure. Before putting in the time to rewrite it, we want to be sure there are users of this feature. If there are no users interested in this feature, then we would like to remove it.  
If your requirements change in the future, we will welcome an RFC to re-implement the feature

If you are interested in using this feature, please reply to this issue. If we do not hear anything by early January, then we shall mark the feature as deprecated and proceed with removing the functionality soon after.

## Proposal
The 'Preserve contents of playing segment when unsynced' functionality is a contribution received a few year ago. At NRK we have never used this, so any knowledge about how it works it from reading the code and remembering prior discussions (I cannot find the PR for this, and it has survived a couple of large refactorings so git history will be hard to follow). We are also unsure if it works fully in current releases, and we believe the original contributor no longer uses this feature.

This setting changes what happens when the currently playing Segment is removed from the NRCS.
The default Sofie behaviour is that when a Segment is deleted from the NRCS, all the Parts will get removed from Sofie, leaving just the current and next PartInstance. Once the playhead leaves the Segment it will finish being removed.
With this setting enabled, the Segment will instead remain with all its Parts and other contents. I am unsure when these will eventually be removed (perhaps also when the playhead leaves the Segment?)

The proposal is simply to remove this feature, as we are not aware of any users of this feature.

## Status

The Sofie Team will evaluate this RFC and open up a discussion about it, usually within a week.

- [x] RFC created
- [ ] Sofie Team has evaluated the RFC
- [ ] A workshop has been planned
- [ ] RFC has been discussed in a workshop
- [ ] A conclusion has been reached, see comments in thread.

**Comment by Jachtv2** [comment]
TV 2 Denmark is probably the contributor, and has the functionality enabled. 
It gives the producer all planned options, if the NRCS segment is accidentally closed. 
It is correct, that the segment is removed once the playhead leaves the segment.

**Comment by Julusian** [comment]
@Jachtv2 thank you for responding. I expect you are correct on the contributor.

To clarify, is this functionality that you are using and intend to keep using?

**Comment by Jachtv2** [comment]
@Julusian TV 2 Denmark intends to keep this functionality, but since we are moving towards https://github.com/tv2/sofie-server, my best guess is, that we are not relying on the outcome of this RFC.

**Comment by Julusian** [comment]
@Jachtv2 yes, it sounds like you don't intend to update the version of sore-core you are using (from R46 to R51), so this won't impact you unless that changes

**Comment by Julusian** [comment]
It sounds like there is no-one who wants this functionality then, so it shall be removed as part of #1108

If that changes, we welcome a PR to reimplement it. You can start this by reverting https://github.com/nrkno/sofie-core/commit/0d54e1facac5e240b1b01c3ba5f9780c3a610ef5 which was the commit that removed the functionality. 
That commit removed a prototype reimplementation which had not been tested and I expect had some edge cases which needed solving. I have concerns about the safety of undeleting Parts, as the same ingest operation could have moved one of those Parts to a different Segment. There could be more issues that need solving too.
If any of this is unclear feel free to reach out
